### DISCUSSION TITLE
[Chapter05] 클래스 책임 분리해보기

### DISCUSSION THREAD

**Discussion by 0sunset0** [cowaine/is-my-code-that-strange#34] [Chapter 04-05 (불변, 응집도)] [answered]
77p에 있는 common 클래스에는 관련 없는 메서드들이 모여 있습니다.

```java

public class Common {

    //세금 포함 금액 계산하기
    static BigDecimal calcAmountIncludingTax(BigDecimal amountExcludingTax, BigDecimal taxRate) {
        return amountExcludingTax.multiply(taxRate);
    }

    //사용자가 이미 탈퇴했다면 true
    static boolean hasResigned(User user) {
       ...
    }


    //상품 주문하기
    static void createOrder(Product product) {
       ...
    }

    static boolean isValidPhoneNumber(String phoneNumber) {
       ...
    }
}
```
이 클래스는 몇 개로 쪼개져야 하고, 클래스 이름은 무엇이 되어야 할까요?
~~클래스는 static 클래스일까요~~

**Comment by snoopingh** [comment, marked as answer]
User의 휴대폰번호라고 가정했을 때, 3개로 나눌 것 같습니다. (User, ContractAmountUtil, Order)
hasResigned, isValidPhoneNumber  : 탈퇴여부나 휴대폰번호는 User가 있어야 존재하는 값이니 User내에 작성
calcAmountIncludingTax : 세금관련된 계산 Util 클래스에 static 메소드로 작성 (객체의 상태에 대한 변경이 없음)
createOrder : 주문과 관련된 클래스에 작성

클래스가 static 이냐는건 어떤 말씀이신지 잘 모르겠네요

**Reply by 0sunset0** [reply]
Inner class를 쓰다보니 헷갈렸네요.. ㅋㅋㅠㅠ

**Reply by firewoody237** [reply]
왜 다 `static`으로 붙여놨을 까요..?
저도 @snoopingh 님처럼 3가지로 나눌 것 같습니다.
`static`은 `calcAmountIncludedTax`에만 있을 것 같아요

**Comment by Irisation23** [comment]
### 체크 포인트

- Inner 클래스로 DTO 만들기의 단점?
